File: 00-code(源代码)/src/com/hi/dhl/algorithms/offer/_18/java/Solution.java
package com.hi.dhl.algorithms.offer._18.java;

import com.hi.dhl.algorithms.model.ListNode;

/**
 * <pre>
 *     author: dhl
 *     date  : 2020/8/5
 *     desc  :
 * </pre>
 */
class Solution {
    public ListNode deleteNode(ListNode head, int val) {
        if (head == null) return head;
        if (head.val == val) return head.next;

        ListNode current = head;
        while (current.next != null) {
            ListNode tmp = current.next;
            if (tmp.val == val) {
                current.next = tmp.next;
                break;
            }
            current = tmp;
        }
        return head;
    }
}

File: 00-code(源代码)/src/com/hi/dhl/algorithms/offer/_18/kotlin/Solution.kt
package com.hi.dhl.algorithms.offer._18.kotlin

import com.hi.dhl.algorithms.model.ListNode

/**
 * <pre>
 *     author: dhl
 *     date  : 2020/8/5
 *     desc  :
 * </pre>
 */

class Solution {
    fun deleteNode(head: ListNode?, `val`: Int): ListNode? {
        if (head == null) return head
        if (head.`val` == `val`) return head.next

        var current = head
        while (current?.next != null) {
            val tmpNode = current.next;
            if (tmpNode.`val` == `val`) {
                current.next = tmpNode.next
                break;
            }
            current = tmpNode
        }
        return head
    }
}